client/my-sites/controller.js

```diff
@@ -494,6 +494,9 @@ export function siteSelection( context, next ) {
 					}
 				} else if ( shouldRedirectToJetpackAuthorize( context, site ) ) {
 					navigate( getJetpackAuthorizeURL( context, site ) );
+				} else if ( 'plugins' === context.path.split( '?' )[ 0 ].split( '/' )[ 1 ] ) {
+					// When path start with /plugins, redirect won't use site as query param. In plugins the param could be a site, plugin or category.
+					page.redirect( sectionify( context.path, siteFragment ) );
```

Comment:
Is there a way to do this without specifying a specific route? I'm wondering if this is overstepping a boundary as this code is quite general / there doesn't seem to be any other specific overrides like this in place already.

Reply:
I made this in order to now broke other pages that uses the `site` query param

Reply:
we explored a couple of ways of doing this, and this was the one with the least impact. Other way it would require wider testing through the app